src/java.naming/share/classes/com/sun/jndi/ldap/AbstractLdapNamingEnumeration.java

```diff
@@ -333,8 +331,11 @@ protected final boolean hasMoreReferrals() throws NamingException {
 
                 } catch (LdapReferralException re) {
 
-                    // record a previous exception
-                    if (errEx == null) {
+                    // record a previous exception and quit if any limit is reached
+                    if (re.getNamingException() instanceof LimitExceededException) {
+                        errEx = re.getNamingException();
+                        break;
+                    } else if (errEx == null) {
```

Comment:
Could we set up a local variable here to avoid calling re.getNamingException() more than once?
e.g. something like:

```
var cause = re.getNamingException();
if (cause instanceof ...) {
```

then use `cause` throughout.

Reply:
Yes, of course, I just called the var `namingException` instead of `cause`. It's done now.

Reply:
Sorry! I have squashed the two commits... I didn't know the system will do it before merging. :disappointed: